hubbot

```diff
@@ -349,20 +361,42 @@ def expire_work_dirs(github_visible_commits):
     # make sure we don't look at too many commits
     github_visible_commits = list(set(github_visible_commits))
     def is_redundant_directory(dirname):
-        if dirname in ["data", "status-images"] or not os.path.isdir(os.path.join(work,dirname)):
-            return False
         for c in github_visible_commits:
             if c in dirname:
                 return False
         return True
-    dirs = sorted (map (lambda e: (os.stat(e).st_mtime,e),
-                        map (lambda e: os.path.join(work,e),
-                             filter (is_redundant_directory, os.listdir(work)))),
-                   lambda x,y: cmp(x[0],y[0]))
+    dirs = filter (is_redundant_directory, list_work_dirs())
     if len(dirs) > keep_work_dirs:
         for d in dirs[0:len(dirs)-keep_work_dirs]:
-            trace("Expiring %s" % d[1])
-            shutil.rmtree(d[1])
+            trace("Expiring %s" % d)
+            shutil.rmtree(d)
+
+def expire_images():
+    keep = { }
+
+    def keep_image(flavor, os_, tag):
+        keep["%s-%s-*-%s-checksum" % (flavor, os_, tag)] = True
+        keep["%s-%s-*-%s.*" % (flavor, os_, tag)] = True
+
+    for d in list_work_dirs():
+        t = os.path.join(d, "test")
+        if os.path.exists(t):
+            for c in filter(lambda e: e.endswith(".conf"), os.listdir(t)):
+                flavor = c[:-5]
+                conf = ast.literal_eval(open(os.path.join(t, c)).read())
```

Comment:
in case there are other .conf files, we should catch exceptions here and continue as if there were no images to keep